halo/attest/keeper/common_internal_test.go

```diff
@@ -0,0 +1,109 @@
+package keeper
+
+import (
+	"testing"
+
+	"github.com/omni-network/omni/halo/attest/testutil"
+	"github.com/omni-network/omni/halo/attest/types"
+	"github.com/omni-network/omni/lib/errors"
+
+	storetypes "cosmossdk.io/store/types"
+	"github.com/cosmos/cosmos-sdk/runtime"
+	sdktestutil "github.com/cosmos/cosmos-sdk/testutil"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	moduletestutil "github.com/cosmos/cosmos-sdk/types/module/testutil"
+	"go.uber.org/mock/gomock"
+)
+
+type mocks struct {
+	skeeper *testutil.MockStakingKeeper
+	voter   *testutil.MockVoter
+	namer   *testutil.MockChainNamer
+}
+
+func mockDefaultExpectations(_ sdk.Context, m mocks) {}
+
+func setupKeeper(t *testing.T, expectations ...func(sdk.Context, mocks)) (*Keeper, sdk.Context) {
+	t.Helper()
+
+	key := storetypes.NewKVStoreKey(types.StoreKey)
+	storeSvc := runtime.NewKVStoreService(key)
+	ctx := sdktestutil.DefaultContext(key, storetypes.NewTransientStoreKey("test_key"))
+	codec := moduletestutil.MakeTestEncodingConfig().Codec
+
+	// gomock initialization
+	ctrl := gomock.NewController(t)
+	m := mocks{
+		skeeper: testutil.NewMockStakingKeeper(ctrl),
+		voter:   testutil.NewMockVoter(ctrl),
+		namer:   testutil.NewMockChainNamer(ctrl),
+	}
+	if len(expectations) == 0 {
+		mockDefaultExpectations(ctx, m)
+	} else {
+		for _, exp := range expectations {
+			exp(ctx, m)
+		}
+	}
+
+	const voteWindow = 1
+	k, err := New(codec, storeSvc, m.skeeper, m.voter, m.namer.ChainName, voteWindow)
+	if err != nil {
+		t.Fatalf("error creating keeper: %v", err)
+	}
+
+	return k, ctx
+}
+
+// dumpTables returns all the items in the atestation and signature tables as slices.
+func dumpTables(t *testing.T, ctx sdk.Context, k *Keeper) ([]*Attestation, []*Signature) {
+	t.Helper()
+	atts, sigs, err := keeperData(ctx, k)
+	if err != nil {
+		t.Fatalf("error fetching keeper tables: %v", err)
+	}
+
+	return atts, sigs
+}
+
+func keeperData(ctx sdk.Context, k *Keeper) ([]*Attestation, []*Signature, error) {
```

Comment:
please pass in `t *testing.T` and then just do `require.NoError` instead of returning errors in test packages.